vmx/src/vm.rs

```diff
@@ -4,6 +4,9 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
+mod host;
```

Comment:
If you really feel it's neccesary to make a submodule, please use the convention of module_name/mod.rs to not have multiple directories levels of related files.

Reply:
I'm a little puzzled with this. Since [rust 2018 edition](https://doc.rust-lang.org/edition-guide/rust-2018/path-changes.html) `mod.rs` files are no longer needed when making submodules. Do we really want to keep the old convention?
If you look into the `hatchery` on the `main` branch you'll see an example where you don't really see nesting either.